functions/Rename-DbaDatabase.ps1

```diff
@@ -0,0 +1,713 @@
+		would end up with this logic: 
+		- database --> no placeholders specified 
+			- dbatools_HR to dbatools_HRARCHIVE
+				- filenames placeholders specified
+					<DBN><FNN> --> current database name + current filename"
+						- dbatools_HR_Data.mdf to dbatools_HRARCHIVEdbatools_HR_Data.mdf
+						- dbatools_HR_Index.mdf to dbatools_HRARCHIVEdbatools_HR_Data.mdf
+						- dbatools_HR_log.ldf to dbatools_HRARCHIVEdbatools_HR_log.ldf
+		Passing this switch, instead, e.g.
+		Rename-DbaDatabase .... -Database "dbatools_HR" -DatabaseName "dbatools_HRARCHIVE" -FileName '<DBN><FNN>' -ReplaceBefore
+		end up with this logic instead: 
+		- database --> no placeholders specified 
+			- dbatools_HR to dbatools_HRARCHIVE
+				- filenames placeholders specified, 
+					<DBN><FNN>, plus -ReplaceBefore --> current database name + replace OLD "upper level" names inside the current filename
+					- dbatools_HR_Data.mdf to dbatools_HRARCHIVE_Data.mdf
+					- dbatools_HR_Index.mdf to dbatools_HRARCHIVE_Data.mdf
+					- dbatools_HR_log.ldf to dbatools_HRARCHIVE_log.ldf
+
+	.PARAMETER Force
+		Kills any open session to be able to do renames.
+
+	.PARAMETER SetOffline
+		Kills any open session and sets the database offline to be able to move files
+	
+	.PARAMETER Move
+		If you want this function to move files, else you're the one in charge of it. 
+		This enables the same functionality as SetOffline, killing open transactions and putting the database
+		offline, then do the actual rename and setting it online again afterwards
+	
+	.PARAMETER DatabaseCollection
+		Internal parameter to be able to accept piped data
+	
+	.PARAMETER Silent
+		Use this switch to disable any kind of verbose messages
+
+	.NOTES
+		Author: niphlod
+		
+		dbatools PowerShell module (https://dbatools.io, clemaire@gmail.com)
+		Copyright (C) 2016 Chrissy LeMaire
+		This program is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, either version 3 of the License, or (at your option) any later version.
+		This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more details.
+		You should have received a copy of the GNU General Public License along with this program.  If not, see <http://www.gnu.org/licenses/>.
+	
+	.LINK
+		https://dbatools.io/Rename-DbaDatabase
+	
+	.EXAMPLE
+	Rename-DbaDatabase -SqlInstance sqlserver2014a -Database HR -DatabaseName HR2
+
+	Renames the HR database to HR2
+
+	.EXAMPLE
+	Rename-DbaDatabase -SqlInstance sqlserver2014a -Database HR -DatabaseName "dbatools_<DBN>"
+
+	Renames the HR database to dbatools_HR
+	
+	.EXAMPLE
+	Rename-DbaDatabase -SqlInstance sqlserver2014a -Database HR -DatabaseName "dbatools_<DBN>_<DATE>"
```

Comment:
Just a discussion point for ease of use...have not had a chance to pull it down to test out code itself...

I can understand the use of the `-DatabaseName` being what the database will be renamed to but any option to making that more understood? Say calling it `-Rename` or `-NewName`....I'd prefer the latter as it just makes it more self-explanatory. We use this in other commands like `Set-DbaAgentJob` and `NewLogin` for `Rename-DbaLogin`.

Reply:
Well seeing the other parameters now though I guess `DatabaseName` is the best option.